app/src/main/java/com/infomaniak/swisstransfer/ui/screen/newtransfer/importfiles/FilesDetailsScreen.kt

```diff
@@ -0,0 +1,108 @@
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package com.infomaniak.swisstransfer.ui.screen.newtransfer.importfiles
+
+import androidx.compose.foundation.layout.Column
+import androidx.compose.foundation.layout.padding
+import androidx.compose.material3.Surface
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.Modifier
+import androidx.hilt.navigation.compose.hiltViewModel
+import com.infomaniak.swisstransfer.R
+import com.infomaniak.swisstransfer.ui.components.FileItemList
+import com.infomaniak.swisstransfer.ui.components.FileUi
+import com.infomaniak.swisstransfer.ui.components.SwissTransferTopAppBar
+import com.infomaniak.swisstransfer.ui.components.TopAppBarButton
+import com.infomaniak.swisstransfer.ui.screen.newtransfer.FilesSize
+import com.infomaniak.swisstransfer.ui.theme.Margin
+import com.infomaniak.swisstransfer.ui.theme.SwissTransferTheme
+import com.infomaniak.swisstransfer.ui.utils.PreviewAllWindows
+
+@Composable
+fun FilesDetailsScreen(
+    filesDetailsViewModel: FilesDetailsViewModel = hiltViewModel<FilesDetailsViewModel>(),
+    fileId: String,
+    navigateToDetails: (String) -> Unit,
+    withSpaceLeft: Boolean,
+    onFileRemoved: ((uuid: String) -> Unit)? = null,
+    onCloseClicked: (() -> Unit),
+    navigateBack: (() -> Unit),
+) {
+    FilesDetailsScreen(
+        title = {
+            //TODO Get fileId detail to get the title
+            "Some Folder"
+        },
+        files = { filesDetailsViewModel.getFilesFromUUID(fileId) },
+        navigateToDetails = navigateToDetails,
+        withSpaceLeft = withSpaceLeft,
+        onFileRemoved = onFileRemoved,
+        onCloseClicked = onCloseClicked,
+        navigateBack = navigateBack,
+    )
+}
+
+@Composable
+private fun FilesDetailsScreen(
+    title: () -> String,
+    files: () -> List<FileUi>,
+    navigateToDetails: (String) -> Unit,
+    withSpaceLeft: Boolean,
+    onFileRemoved: ((uuid: String) -> Unit)? = null,
+    onCloseClicked: (() -> Unit),
+    navigateBack: (() -> Unit),
+) {
+    val filesToDisplay = files()
+    Column {
+        SwissTransferTopAppBar(
+            title = title(),
+            titleRes = R.string.importFilesScreenTitle,
+            navigationMenu = TopAppBarButton.backButton { navigateBack() },
+            TopAppBarButton.closeButton { onCloseClicked() },
+        )
+
+        FilesSize(filesToDisplay, withSpaceLeft)
+        FileItemList(
+            modifier = Modifier.padding(horizontal = Margin.Medium),
+            files = filesToDisplay,
+            isRemoveButtonVisible = onFileRemoved != null,
+            isCheckboxVisible = false,
+            isUidChecked = { false },
+            setUidCheckStatus = { _, _ -> },
+            onRemoveUid = { onFileRemoved?.invoke(it) },
+            onClick = { navigateToDetails(it) }
+        )
+    }
+}
```

Comment:
I don't know to what extent this part is optimized for recompositions. If you haven't already, maybe just make sure it's optimized with @sirambd or by checking the recompositions in the layout inspector. And if you do ask him, I'd be interested to listen to the explanation as well btw